File: go-server/internal/modules/sample/application/queries/list_samples.go
package queries

import (
	"context"

	"cariban/go-server/internal/modules/sample/application"
	"cariban/go-server/internal/modules/sample/domain"
)

type ListSamplesQuery struct {
	repository domain.SampleRepository
}

func NewListSamplesQuery(repository domain.SampleRepository) *ListSamplesQuery {
	return &ListSamplesQuery{repository: repository}
}

func (q *ListSamplesQuery) Execute(ctx context.Context, req application.ListSamplesRequest) (application.ListSamplesResponse, error) {
	limit := req.Limit
	if limit <= 0 {
		limit = 10
	}
	offset := req.Offset
	if offset < 0 {
		offset = 0
	}

	entities, total, err := q.repository.FindAll(ctx, limit, offset)
	if err != nil {
		return application.ListSamplesResponse{}, err
	}

	items := make([]application.SampleItem, 0, len(entities))
	for _, entity := range entities {
		items = append(items, application.SampleItem{
			ID:        entity.ID,
			Name:      entity.Name.String(),
			CreatedAt: entity.CreatedAt,
		})
	}

	return application.ListSamplesResponse{
		Items:  items,
		Total:  total,
		Limit:  limit,
		Offset: offset,
	}, nil
}

File: fastapi-server/src/modules/sample/application/queries/list_samples.py
"""sample モジュールのサンプル一覧クエリ。"""

from src.modules.sample.domain import SampleRepository

from ..dto import ListSamplesRequest, ListSamplesResponse, SampleItem


class ListSamplesQuery:
    """サンプル一覧クエリ

    リポジトリを使用してページネーション付きのサンプル一覧を取得します。
    """

    def __init__(self, repository: SampleRepository) -> None:
        """リポジトリでクエリを初期化

        Args:
            repository: データアクセス用のサンプルリポジトリ
        """
        self._repository = repository

    async def execute(self, request: ListSamplesRequest) -> ListSamplesResponse:
        """サンプル一覧クエリを実行

        Args:
            request: ページネーションパラメータを含む一覧リクエスト DTO

        Returns:
            アイテムと総数を含む一覧レスポンス DTO
        """
        entities, total = await self._repository.find_all(
            limit=request.limit,
            offset=request.offset,
        )

        items = [
            SampleItem(
                id=entity.id,
                name=str(entity.name),
                created_at=entity.created_at,
            )
            for entity in entities
        ]

        return ListSamplesResponse(
            items=items,
            total=total,
            limit=request.limit,
            offset=request.offset,
        )
